extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -1,93 +1,156 @@
 import React, { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { BigNumber } from "bignumber.js";
+import { Field, Form, Formik, FieldProps } from "formik";
+import { Asset } from "stellar-sdk";
 
 import { Button, Select } from "@stellar/design-system";
 
-import { AccountBalancesInterface } from "@shared/api/types";
-
 import { navigateTo } from "popup/helpers/navigate";
 import { ROUTES } from "popup/constants/routes";
-
+import { FormRows } from "popup/basics/Forms";
 import { PopupWrapper } from "popup/basics/PopupWrapper";
-
 import { BackButton } from "popup/basics/BackButton";
+import {
+  transactionSubmissionSelector,
+  saveAmount,
+  saveAsset,
+} from "popup/ducks/transactionSubmission";
+import { AccountDoesntExistWarning } from "../SendTo";
 
 import "../styles.scss";
 
-export const SendAmount = ({
-  amount,
-  setAmount,
-  asset,
-  setAsset,
-  accountBalances,
-}: {
-  amount: string;
-  setAmount: (state: string) => void;
-  asset: string;
-  setAsset: (state: string) => void;
-  accountBalances: AccountBalancesInterface;
-}) => {
-  const [selectedAsset, setSelectedAsset] = useState({
-    code: "",
+const baseReserve = new BigNumber(1);
+
+export const SendAmount = () => {
+  const dispatch = useDispatch();
+  const { accountBalances, destinationBalances, transactionData } = useSelector(
+    transactionSubmissionSelector,
+  );
+  const { amount, asset } = transactionData;
+
+  const [assetInfo, setAssetInfo] = useState({
+    code: Asset.native().code,
     balance: "0",
-    canonical: "",
+    canonical: Asset.native().toString(),
   });
 
   useEffect(() => {
     if (accountBalances.balances) {
-      setSelectedAsset({
+      setAssetInfo({
         code: accountBalances.balances[asset].token.code,
         balance: accountBalances.balances[asset].total.toString(),
         canonical: asset,
       });
-    } else {
-      setSelectedAsset({
-        code: "XLM",
-        balance: "0",
-        canonical: "native",
-      });
     }
   }, [asset, accountBalances]);
 
+  const handleContinue = (values: { amount: string }) => {
+    dispatch(saveAmount(values.amount));
+    dispatch(saveAsset(assetInfo.canonical));
+    navigateTo(ROUTES.sendPaymentSettings);
+  };
+
+  const handleAssetSelect = (e: React.ChangeEvent<any>) => {
```

Comment:
Sorry I didn't catch this before: You can pass the input type to `React.ChangeEvent` instead of `any`

 you can use `React.ChangeEvent<HTMLSelectElement>` here, for example

Reply:
ah cool  https://github.com/stellar/freighter/pull/400/commits/5216670a4a3675872ea41c608f817dbd0fe6d2e2